### PR TITLE
[it] Update links to point localized pages

### PR DESCRIPTION
**Pull request by jihoon-seo**
~I deliberately set the target branch to `main`, since~ this is kind of hotfix, not a translating job. 😊

Edit: this PR now supersedes the PR #833, so merging this PR closes #833.

### CODE REVIEW

File: content/it/_index.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ Il Glossario Cloud Native è un progetto guidato dal Business Value Subcommittee
 
 L'invito a suggerire modifiche, aggiunte e migliorie al Glossario Cloud Native è aperto a chiunque abbia voglia di contribuire. Il processo che adottiamo per lo sviluppo e l'arricchimento di questo dizionario condiviso è guidato dalla community e governato dalla CNCF. Il Glossario fornisce una piattaforma vendor-neutral per organizzare un vocabolario condiviso riguardante le tecnologie cloud native. 
 
-Chiunque voglia contribuire può aprire una issue o una pull request su GitHub. Leggi la sezione [Come Contribuire](/contribuire/) per maggiori informazioni e assicurati di seguire le indicazioni della [Style Guide](/style-guide/).
+Chiunque voglia contribuire può aprire una issue o una pull request su GitHub. Leggi la sezione [Come Contribuire](/it/contribute/) per maggiori informazioni e assicurati di seguire le indicazioni della [Style Guide](/it/style-guide/).
```
Comment (jihoon-seo) [inline content/it/_index.md:16]: Dear Italian contributors,
which one would be better here? ("Style Guide" vs. "Guida di Stile")

```suggestion
Chiunque voglia contribuire può aprire una issue o una pull request su GitHub. Leggi la sezione [Come Contribuire](/it/contribute/) per maggiori informazioni e assicurati di seguire le indicazioni della [Guida di Stile](/it/style-guide/).
```

Comment (annalisag-spark) [reply content/it/_index.md:16]: Ciao @jihoon-seo! They are both correct, but we can go for Guida di Stile to be more consistent :)
Thank you!

Comment (jihoon-seo) [reply content/it/_index.md]: @annalisag-spark Okay. 😊

Do you have suggestion for this too?
https://glossary.cncf.io/it/style-guide/#:~:text=style%20guide%20di%20base%20del%20repository%20CNCF
> Il Glossario Cloud Native rispetta la [style guide di base](https://github.com/cncf/foundation/blob/master/style-guide.md) del repository CNCF

Comment (annalisag-spark) [reply content/it/_index.md]: Yep! Changed into Guida di Stile here too!
Hope I followed the right steps to commit the change and open a PR 😅

Thank you for your meticulous review.

**Comment by jihoon-seo** [comment]
Dear Italian contributors,
please review the commit 7fb77dc, as well as the entire PR!

**Review by annalisag-spark — APPROVED** [review]

**Comment by seokho-son** [comment]
@jihoon-seo I hope to recommend l10n teams to use dev-xx branch if possible.
I think changing links to localized documents seems not critical bug to be fixed quickly.

If we keep updating localized files in main branch, l10n teams will eventually meet merge conflicts. (also, l10n contributors can confuse target branch)
So, I personally prefer not using main branch directly if the problem to be fixed is not too critical.

What do you think?

**Comment by jihoon-seo** [comment]
@seokho-son As per your comment, I changed the target branch as `dev-it`. 😊

**Comment by seokho-son** [comment]
Thanks @jihoon-seo .
From now, this PR is upto IT team's reviewers :)

(Maintainers can approve this PR, but I hope a dev-xx branch is managed by each team if possible)

**Review by sistella — APPROVED** [review]

**Pull request merged**